pkg/uploadserver/uploadserver_test.go

```diff
@@ -204,209 +200,175 @@ func replaceAsyncProcessorFunc(replacement func(io.ReadCloser, string, string, s
+		})
 
-		if status := rr.Code; status != http.StatusServiceUnavailable {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusServiceUnavailable)
-		}
 	})
-}
-func TestCompletedConflict(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newRequest(t)
 
-		rr := httptest.NewRecorder()
+	It("Completed conflict", func() {
+		withProcessorSuccess(func() {
+			req := newRequest()
+
+			rr := httptest.NewRecorder()
 
-		server := newServer()
-		server.done = true
-		server.ServeHTTP(rr, req)
+			server := newServer()
+			server.done = true
+			server.ServeHTTP(rr, req)
 
-		if status := rr.Code; status != http.StatusConflict {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusConflict)
-		}
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusConflict))
+		})
 	})
-}
 
-func TestCompletedConflictAsync(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newAsyncRequest(t)
+	It("Completion conflict, async", func() {
+		withProcessorSuccess(func() {
+			req := newAsyncRequest()
 
-		rr := httptest.NewRecorder()
+			rr := httptest.NewRecorder()
 
-		server := newServer()
-		server.done = true
-		server.ServeHTTP(rr, req)
+			server := newServer()
+			server.done = true
+			server.ServeHTTP(rr, req)
 
-		if status := rr.Code; status != http.StatusConflict {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusConflict)
-		}
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusConflict))
+		})
 	})
-}
-func TestSuccess(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newRequest(t)
 
-		rr := httptest.NewRecorder()
+	It("Success", func() {
+		withProcessorSuccess(func() {
+			req := newRequest()
 
-		server := newServer()
-		server.ServeHTTP(rr, req)
+			rr := httptest.NewRecorder()
 
-		if status := rr.Code; status != http.StatusOK {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusOK)
-		}
+			server := newServer()
+			server.ServeHTTP(rr, req)
+
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusOK))
+		})
 	})
-}
 
-func TestSuccessAsync(t *testing.T) {
-	withAsyncProcessorSuccess(func() {
-		req := newAsyncRequest(t)
+	It("Success, async", func() {
+		withAsyncProcessorSuccess(func() {
+			req := newAsyncRequest()
 
-		rr := httptest.NewRecorder()
+			rr := httptest.NewRecorder()
 
-		server := newServer()
-		server.ServeHTTP(rr, req)
+			server := newServer()
+			server.ServeHTTP(rr, req)
 
-		if status := rr.Code; status != http.StatusOK {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusOK)
-		}
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusOK))
+		})
 	})
-}
 
-func TestSuccessHeadAsync(t *testing.T) {
-	withAsyncProcessorSuccess(func() {
-		req := newAsyncHeadRequest(t)
+	It("Success HEAD, async", func() {
+		withAsyncProcessorSuccess(func() {
```

Comment:
Only difference between this one and the one above is the request type, can use a table with the request as a parameter to reduce code.